### PR TITLE
link to examples in concepts pages

### PR DESCRIPTION
**Pull request by jamiedemaria**
### Summary & Motivation
Adds links to the examples of concepts being used in the hackernews project #6829

Some concepts that i couldn't find examples for:
- [ ] dynamic outputs
- [ ] nested graphs 
- [ ] input loading
- [ ] hooks 
- [ ] the schedule example in hacker news is really a partition example

### CODE REVIEW

File: docs/content/concepts/resources.mdx [hunk 1]
```diff
@@ -260,3 +260,10 @@ When constructing a job that includes that op, we provide the resource `client`,
 def connect():
     get_client()
 ```
+
+## See it in action
+For more examples of resources, check out the following in our [Hacker News example](https://github.com/dagster-io/dagster/tree/master/examples/hacker_news):
+* [Hacker News resource](https://github.com/dagster-io/dagster/blob/master/examples/hacker_news/hacker_news/resources/hn_resource.py)
+* [dbt asset resource](https://github.com/dagster-io/dagster/blob/master/examples/hacker_news/hacker_news/resources/dbt_asset_resource.py)
```
Comment (sryza) [inline docs/content/concepts/resources.mdx:267]: I think this one is a bit confusing and might be better left out.

**Review by sryza — COMMENTED** [review]
This is great.

If you look at the READMEs of the nyt example and the bollinger example, they list features that those examples use.  It could be useful to add the inverse mapping of those in here as well.

File: docs/content/concepts/assets/software-defined-assets.mdx [hunk 2]
```diff
@@ -412,3 +412,11 @@ def my_asset(context):
 ## Further Reading
 
 Interested in learning more about software-defined assets and working through a more complex example? Check out our [guide on software-defined assets](/guides/dagster/software-defined-assets) and our [example project](https://github.com/dagster-io/dagster/tree/master/examples/modern_data_stack_assets) that integrates software-defined assets with other Modern Data Stack tools.
+
+## See it in action
+For more examples of jobs, check out the following in our [SDA Hacker News example](https://github.com/dagster-io/dagster/tree/master/examples/hacker_news_assets):
```
[the remark this answers is not in the document]
Comment (jamiedemaria) [reply docs/content/concepts/assets/software-defined-assets.mdx:417]: yup, missed it when i copied over

File: docs/content/concepts/assets/software-defined-assets.mdx [hunk 3]
```diff
@@ -412,3 +412,11 @@ def my_asset(context):
...
+* [Defining an asset](https://github.com/dagster-io/dagster/blob/master/examples/hacker_news_assets/hacker_news_assets/assets/activity_forecast.py)
```
[the remark this answers is not in the document]
Comment (jamiedemaria) [reply docs/content/concepts/assets/software-defined-assets.mdx:418]: yeah i was thinking about that as i was writing this out. looks like there are some packages that will test the links in md files, i can investigate and see if any of them are promising

File: docs/content/concepts/resources.mdx (hunk 1 above)
Comment (jamiedemaria) [reply docs/content/concepts/resources.mdx:267]: just confirming, you think the dbt asset resource is confusing?

Comment (sryza) [reply docs/content/concepts/resources.mdx:267]: exactly

File: docs/content/concepts/assets/software-defined-assets.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (sryza) [reply docs/content/concepts/assets/software-defined-assets.mdx:418]: fwiw I believe there's already something we use that checks links that are internal in the docs site

[the remark this answers is not in the document]
Comment (jamiedemaria) [reply docs/content/concepts/assets/software-defined-assets.mdx:418]: oh cool ok i'll track that down

[the remark this answers is not in the document]
Comment (jamiedemaria) [reply docs/content/concepts/assets/software-defined-assets.mdx:418]: found where we check internal links, and adding a new test for external links should be straightforward except for the part that actually tests the links. i'll need to modify a function like [this](https://github.com/dagster-io/dagster/blob/6e502070a981b0db8e0cce0ea2eb7f41999aa025/docs/next/__tests__/mdxInternalLinks.test.ts#L95) to validate the external link, but not sure how to do that yet. searching around for examples of other people doing something similar

[the remark this answers is not in the document]
Comment (sryza) [reply docs/content/concepts/assets/software-defined-assets.mdx:418]: something that might make it easier is that the link isn't fully "external" - it points to stuff that's in the repo

[the remark this answers is not in the document]
Comment (jamiedemaria) [reply docs/content/concepts/assets/software-defined-assets.mdx]: yeah that was a good insight. have a test in and it looks like it's working

**Review by sryza — APPROVED** [review]
This is quite awesome

**Comment by yuhan** [comment]
omg this is so good - i've been cleaning up the docs infra lately so have been slow on reviews, so apologies for the late response. 

i think this is a fantastic idea and lets land this!! in the next iteration, maybe we could move this section a bit higher up and make a standalone ui component to promote this section - i believe it's gonna be very helpful to readers, i'll bring to josh and erin for design review.

**Pull request merged**